MailSim/MailSim/ProvidersOM/MailStoreProviderOM.cs

```diff
@@ -189,9 +190,50 @@ public IAddressBook GetGlobalAddressList()
             if (_outlook != null && !_keepOutlookRunning)
             {
                 Console.WriteLine("Exiting Outlook");
- 
+
                 ((Outlook._Application)_outlook).Quit();
             }
         }
+
+
+        /// <summary>
+        /// Finds the Global Address List associated with the MailStore
+        /// </summary>
+        /// <param name="folderPath">parameter for FindFolder</param>
+        /// <returns>IMailFolder for the target of the event</returns>
+        /// In this case the Folder element is a path in Outlook. Each component of the path separated by '\'.
+        /// The first or default folder in the path, can be preceded by "\\" or nothing. If it's the only part of
+        /// the path, then it MUST be one of the default OL folders (see the schema for the EventMonitor operation). 
+        public IMailFolder FindFolder(string folderPath)
+        {
+            IMailFolder folder;
+            string backslash = @"\";
+            try
+            {
+                if (folderPath.StartsWith(@"\\"))
+                {
+                    folderPath = folderPath.Remove(0, 2);
+                }
+                String[] folders =
+                    folderPath.Split(backslash.ToCharArray());
+                folder =
+                    GetDefaultFolder(folders[0]);
+
+                if (folder != null)
+                {
+                    for (int i = 1; i <= folders.GetUpperBound(0); i++)
```

Comment:
.Length is fine since folders is a simple one dimension array. I will use that. 
However, foreach will go through all the elements. In this case, we want to skip the first element. So we should keep the for(...) construct.